src/js/service/PiskelApiService.js

```diff
@@ -0,0 +1,157 @@
+/* @file Service for managing communication with an external application
+ * when Piskel is embedded in an iframe. */
+(function () {
+  var ns = $.namespace('pskl.service');
+  var MessageType = PiskelApi.MessageType;
+
+  /**
+   * Service for managing communication with an external application when
+   * Piskel is embedded in an iframe. Receives from and sends to the PiskelApi
+   * class; see PiskelApi.js for more information.
+   *
+   * @param {!PiskelController} piskelController
+   * @param {!ImportService} importService
+   * @constructor
+   */
+  ns.PiskelApiService = function (
+      piskelController, importService) {
+    /**
+     * The import service
+     * @private {!PiskelController}
+     */
+    this.piskelController_ = piskelController;
+
+    /**
+     * The import service we'll rely on to load animations when they are
+     * selected in the outer UI.
+     * @private {!ImportService}
+     */
+    this.importService_ = importService;
+
+    /**
+     * Reference to the parent window (in an embedded context) used to send
+     * messages out of our frame.
+     * @private {Window}
+     */
+    this.parentWindow_ = null;
+
+    /**
+     * Which origin we will receive messages from.  Set on init().
+     * @private {string}
+     */
+    this.allowedOrigin_ = null;
+  };
+
+  /**
+   * Initialize message service.  Causes service to start listening for messages
+   * on window.
+   * @param {!Window} piskelWindow
+   */
+  ns.PiskelApiService.prototype.init = function (piskelWindow) {
+    this.parentWindow_ = piskelWindow.parent;
+
+    // We only accept messages sent from the origin hosting Piskel.
+    this.allowedOrigin_ = piskelWindow.location.origin;
+    piskelWindow.addEventListener('message', this.receiveMessage_.bind(this));
+
+    // For implementing auto-save: On a save state event, we should notify
+    // the parent app that the animation has changed.
+    $.subscribe(Events.PISKEL_SAVE_STATE, this.onSaveStateEvent.bind(this));
+
+    this.log('Initialized.');
+  };
+
+  // Debug logging utility - should remove at some point.
+  var VERBOSE = false;
```

Comment:
not sure how temporary this is meant to be, but if it's gonna be around for a bit I might put it as a member of the class rather than a naked variable like this.